addons/mail/static/tests/message/message.test.js

```diff
@@ -1157,6 +1157,52 @@ test("prevent attachment delete on non-authored message in channels", async () =
     await contains(".o-mail-AttachmentImage div[title='Remove']", { count: 0 });
 });
 
+test("prevent attachment delete on non-authored message in threads", async () => {
+    // admin would always be able to delete
+    const pyEnv = await startServer();
+    patchWithCleanup(pyEnv["res.users"], {
+        _init_store_data() {
+            const res = super._init_store_data(...arguments);
+            res.Store.self.isAdmin = false;
+            return res;
+        },
+    });
+    const partnerId = pyEnv["res.partner"].create({});
+    pyEnv["mail.message"].create({
+        attachment_ids: [
+            [
+                0,
+                0,
+                {
+                    mimetype: "image/jpeg",
+                    name: "BLAH",
+                    res_id: partnerId,
+                    res_model: "res.partner",
+                },
+            ],
+            [
+                0,
+                0,
+                {
+                    mimetype: "image/png",
+                    name: "BLEH",
+                    res_id: partnerId,
+                    res_model: "res.partner",
+                },
+            ],
+        ],
```

Comment:
Could use `Command.create(data)` rather than `[0, 0, data]`

Reply:
Fair, could probably also have used a list comprehension to be honest in hindsight